src/serial/transport_id.cpp

```diff
@@ -0,0 +1,66 @@
+#include "adm/serial/transport_id.hpp"
+#include <sstream>
+#include "adm/detail/id_parser.hpp"
+
+namespace adm {
+
+  // ---- Defaults ---- //
+  const TransportIdValue TransportId::valueDefault_ = TransportIdValue(0);
+
+  // ---- Getter ---- //
+  TransportIdValue TransportId::get(
+      detail::ParameterTraits<TransportIdValue>::tag) const {
+    return boost::get_optional_value_or(value_, valueDefault_);
+  }
+
+  // ---- Has ---- //
+  bool TransportId::has(detail::ParameterTraits<TransportIdValue>::tag) const {
+    return true;
+  }
+
+  // ---- isDefault ---- //
+  bool TransportId::isDefault(
+      detail::ParameterTraits<TransportIdValue>::tag) const {
+    return value_ == boost::none;
+  }
+
+  // ---- Setter ---- //
+  void TransportId::set(TransportIdValue value) { value_ = value; }
+
+  // ---- Unsetter ---- //
+  void TransportId::unset(detail::ParameterTraits<TransportIdValue>::tag) {
+    value_ = boost::none;
+  }
+
+  // ---- Operators ---- //
+  bool TransportId::operator==(const TransportId& other) const {
+    return get<TransportIdValue>() == other.get<TransportIdValue>();
+  }
+
+  bool TransportId::operator!=(const TransportId& other) const {
+    return !(*this == other);
+  }
+
+  bool TransportId::operator<(const TransportId& other) const {
+    return formatId(*this) < formatId(other);
```

Comment:
should use `OptionalComparator`

Reply:
0c2051e4753b11dc72ddf7256939a63b8df2f4a9